### PR TITLE
Determine compress algorithm by checking magic bytes

### PR DESCRIPTION
**Pull request by guillaumerose**
This allows us to change the format of crcbundle to what we want.

### CODE REVIEW

**Comment by cfergeau** [comment]
Did you consider using a dedicated go module for this content sniffing? https://stackoverflow.com/questions/29838185/how-to-detect-additional-mime-type-in-golang (not saying this should be done, merely asking :)

**Comment by guillaumerose** [comment]
Very good point! I was too happy to write it myself. It's better now :)

File: pkg/extract/extract_test.go [hunk 1]
```diff
@@ -9,21 +9,59 @@ import (
...
+	}
+	filteredFiles fileMap = map[string]string{
+		"/a/b/c.txt": "ccc",
+	}
+	archives = []string{
+		"test.tar",
+		"test.tar.gz",
+		"test.zip",
+		"test.tar.xz",
+	}
```
Comment (praveenkumar) [inline pkg/extract/extract_test.go:32]: During this unit test can we also put a `crcbundle` tarball file in the testdata folder just to make sure it actually able to detect it as a tarball?

Comment (guillaumerose) [reply pkg/extract/extract_test.go:32]: Yes this is exactly the second test https://github.com/code-ready/crc/pull/1428/files#diff-92bf4f1e87f4458b933a72c9b54411d5R42

crcbundle can be any type now and it checks for it.

File: pkg/extract/extract_test.go [hunk 2]
```diff
@@ -9,21 +9,59 @@ import (
...
+func TestUnCompressBundle(t *testing.T) {
+	dir, err := ioutil.TempDir("", "bundles")
+	require.NoError(t, err)
+	defer os.RemoveAll(dir)
+
+	bundle := filepath.Join(dir, "test.crcbundle")
+	for _, archive := range archives {
+		require.NoError(t, crcos.CopyFileContents(filepath.Join("testdata", archive), bundle, 0600))
+		assert.NoError(t, testUncompress(bundle, nil, files))
+		assert.NoError(t, testUncompress(bundle, fileFilter, filteredFiles))
```
Comment (praveenkumar) [inline pkg/extract/extract_test.go:51]: :+1:

**Pull request merged**